src/node/history.h

```diff
@@ -260,18 +261,21 @@ namespace ccf
       kv::Consensus::SignableTxIndices commit_txid_,
       kv::Store& store_,
       T& replicated_state_tree_,
+      SpinLock& state_lock_,
```

Comment:
Whose lock is this?

Reply:
This is the state lock which protected replicated_state_tree_ this is a race condition that exists in multi-threading that exists before this PR.